tests/templates/test_embeddings.py

```diff
@@ -421,21 +421,24 @@ def circuit(x=None):
         res = circuit(x=features[:n_subsystems])
         assert np.allclose(res, target, atol=tol, rtol=0)
 
-    @pytest.mark.parametrize('features, weights, target', [([np.pi/2, 0], [[np.pi, 0, 0]], [-1, 1]),
-                                                           ([0, np.pi/2], [[np.pi, 0, 0]], [1, -1]),
-                                                           ([np.pi/2, np.pi/2], [[np.pi, 0, 0]], [-1, -1])])
-    def test_qaoa_embedding_state_features_and_zz(self, features, weights, target, tol):
-        """Checks the state produced by QAOAEmbedding() is correct if the features and weights are nonzero.
+    @pytest.mark.parametrize('weights, target', [([[np.pi, 0, 0], [0, 0, 0]], [1, 1]),
+                                                           ([[np.pi/2, 0, 0], [0, 0, 0]], [0, 0]),
+                                                           ([[0, 0, 0], [0, 0, 0]], [-1, -1])])
+    def test_qaoa_embedding_state_features_and_zz(self, weights, target, tol):
+        """Checks the output if the features and entangler weights are nonzero.
         Uses RY local fields."""
 
         dev = qml.device('default.qubit', wires=2)
 
         @qml.qnode(dev)
         def circuit(x=None):
-            QAOAEmbedding(features=x, weights=weights, wires=range(2), local_field='Y')
+            QAOAEmbedding(features=x, weights=weights, wires=range(2))
+            # Undo the final X block
+            qml.RX(-np.pi/2, wires=0)
+            qml.RX(-np.pi/2, wires=1)
             return [qml.expval(qml.PauliZ(i)) for i in range(2)]
 
-        res = circuit(x=features)
+        res = circuit(x=[np.pi/2, np.pi/2])
```

Comment:
How come only one of the features was tested?

Reply:
The feature vector in this test is only there to "expose" the action of the ZZ, and a Hadamard is perfect for doing so.

The features itself are tested in ``test_qaoa_embedding_state_zero_weights``...